Setup updates

Sisanu opened the pull request:
- Document the steps in a usable order, explain why it should be run in that specific order
- Fix the main site setup theme, which is used as a package and it requires the package name not the folder name
- Additional grouping of env variables, network/multi-site variables, other helpers

pederan commented on .env.example:
```diff
@@ -7,20 +7,41 @@
-# Generate your keys here: https://roots.io/salts.html
+### Network Setup
+#WP_ALLOW_MULTISITE=true # optional
+#IS_MULTISITE=1 # optional: 0 = single site, 1 = multisite
```
@Sisanu This is not used anywhere and can be removed. `WP_ALLOW_MULTISITE` decides if multi site and optional `SUBDOMAIN_INSTALL` for sub dir or sub domain

pederan commented on .env.example:
```diff
@@ -7,20 +7,41 @@
+### Network Setup
+#WP_ALLOW_MULTISITE=true # optional
+#IS_MULTISITE=1 # optional: 0 = single site, 1 = multisite
+#SUBDOMAIN_INSTALL=false # optional: set up multisite with sub directory or sub domain, only used if IS_MULTISITE=1
```
@Sisanu Change comment to : # optional: set up multisite with sub directory or sub domain, only used if WP_ALLOW_MULTISITE=true

pederan commented on INSTALLATION.md:
```diff
@@ -25,7 +25,12 @@ Themes, plugins, mu-plugins etc. is now structured under packages and built into
+4. `cd app` go inside the app folder
+5. `rm -rf public` remove the default public folder
+6. `git clone git@github.com:DekodeInteraktiv/{YOUR_PROJECT} .` clone the project into the current folder (the . is important)
+7. `cp .env.example .env` copy the environment example file (no not rename it, as it will show as git differences)
```
Please fix typo: `(no not rename it` -> `(not rename it`

Sisanu commented:
Thanks for the feedback, the changes are done. @pederan please review when you have time. 
Thank you!

pederan commented:
> Thanks for the feedback, the changes are done. @pederan please review when you have time.
> Thank you!

@Sisanu I will. It is the naming of the theme I think is confusing. Your code is correct, it is just that the package folder should be named as the composer package name and also the style.css should have the same Theme Name. I think we should change the folder name to `dekode-theme` and Theme Name to `Dekode Theme`. What do you think?

Sisanu commented:
> > Thanks for the feedback, the changes are done. @pederan please review when you have time.
> > Thank you!
> 
> @Sisanu I will. It is the naming of the theme I think is confusing. Your code is correct, it is just that the package folder should be named as the composer package name and also the style.css should have the same Theme Name. I think we should change the folder name to `dekode-theme` and Theme Name to `Dekode Theme`. What do you think?

The thing as far as I understand, is that regardless of the project name, the theme composer file will still state "dekode/dekode-theme" as the theme name, and in the project composer file we add it exactly the same as a dependency. Since Alex recommended that we rename the `theme-base` (that comes from the project base) to `project-theme` for example, then it means the tools/local script will not work anymore at all, since we will have to activate the theme by using the exact package name, and we do not want to change it for every project, right? For consistency, at least the default theme that comes with the project base should be activated or possible to activate, and the wp-cli command works for `dekode-theme` reference only. Does it make any sense?

zpaceinvader commented:
they should either both be called dekode-theme or theme-base. Maybe go for dekode theme. However it should be renamed regardless on new projects.

pederan commented:
> they should either both be called dekode-theme or theme-base. Maybe go for dekode theme. However it should be renamed regardless on new projects.

@Sisanu So rename folder to `dekode-theme` and in style.css set `Theme Name: Dekode Theme` so we keep it to one name. And yes, needs to be renamed in the beginning of the project.

Sisanu commented:
@pederan @zpaceinvader, the theme rename was done. Also, I added the multisite sh script and updated the install details with this, as in most of the cases when we have to deal with the network setup it seems to be difficult to figure out all the steps. Please review that too. Thank you!

pederan commented:
> @pederan @zpaceinvader, the theme rename was done. Also, I added the multisite sh script and updated the install details with this, as in most of the cases when we have to deal with the network setup it seems to be difficult to figure out all the steps. Please review that too. Thank you!

@Sisanu Great! The linting failed. I think you have to run `composer update` to update the .lock file with correct dependencies. Also push that file.

Sisanu commented:
@pederan, you were right, pushing the lock file fixed it. I think this is all set for now, please review.

pederan reviewed (approved):
👍
